packages/webpack/web-webpack-plugin/src/index.ts

```diff
@@ -57,6 +60,31 @@ export class WebWebpackPlugin {
           });
           return encodeOptions;
         });
+        compiler.hooks.afterEmit.tap(WebWebpackPlugin.name, (compilation) => {
+          this.assets2Delete.forEach((asset) => {
+            compilation.deleteAsset(asset);
+            if (compilation.outputOptions.path) {
+              const files2Delete = [
+                asset,
+              ];
+              if (asset.endsWith('.js')) {
+                files2Delete.push(asset + '.map');
+              }
+              for (const file of files2Delete) {
+                fs.rmSync(
```

Comment:
Emitting the asset to disk first and then manually deleting it seems inefficient.

Reply:
Yes, actually the right way to consume our sourcemap is to read source map from memory using `compiler`.

But the plugin user used now are using fs to find all sourcemap files.

<img width="1103" alt="image" src="https://github.com/user-attachments/assets/d8e1b450-e26f-424d-9cc5-94e3a8f16602" />